inc/saiswitch.h

```diff
@@ -2787,6 +2787,22 @@ typedef enum _sai_switch_attr_t
      */
     SAI_SWITCH_ATTR_ARS_PROFILE,
 
+    /**
+     * @brief Available IP Next hop group member entries
+     *
+     * @type sai_uint32_t
+     * @flags READ_ONLY
+     */
+    SAI_SWITCH_ATTR_AVAILABLE_IP_NEXT_HOP_GROUP_MEMBER_ENTRY,
+
+    /**
+     * @brief Available number of VOQ in the VOQ System (chassis)
+     *
+     * @type sai_uint32_t
+     * @flags READ_ONLY
+     */
+    SAI_SWITCH_ATTR_AVAILABLE_SYSTEM_VOQS,
```

Comment:
@JaiOCP 

Can you please see if we can use the below for the AVAILABLE\_VOQS instead of using the switch attribute:

// Available Fabric Ports  
sai\_attribute\_t attr\[1\];  
attr\[0\].id = SAI\_QUEUE\_ATTR\_TYPE;  
attr\[0\].value = SAI\_QUEUE\_TYPE\_UNICAST\_VOQ; // or SAI\_QUEUE\_TYPE\_MULTICAST\_VOQ  
sai\_object\_type\_get\_availability(\<switch\_id>, SAI\_OBJECT\_TYPE\_QUEUE, 1,  
attr, &count)

Reply:
Hi Ravi,

I did some investigation on this and there are two kind of architectures.
1. Where there are dedicated unicast and cast queues. In this case your suggestion is valid and should be used.
2. Other where there is common pool of VoQs and can be configured as mcast or ucast. For this we will need the new attribute.